src/olympia/addons/admin.py

```diff
@@ -194,9 +194,22 @@ def change_view(self, request, object_id, form_url='', extra_context=None):
                     url += '?' + request.GET.urlencode()
                 return http.HttpResponsePermanentRedirect(url)
 
-        return super(AddonAdmin, self).change_view(
-            request, object_id, form_url, extra_context=None,
+        response = super(AddonAdmin, self).change_view(
+            request, object_id, form_url, extra_context=extra_context
         )
+        if request.method == 'GET':
```

Comment:
I know I'm the one who suggested it, but maybe there is a better way to do this. You would need to double-check, but maybe you can just use `response.status_code` instead ? It's a minor detail, but it would mean the links wouldn't suddenly disappear when you're doing a `POST` but there is an error to be shown so the change form is displayed back to the user.

Reply:
ah, right..